widget_driver_generator/lib/src/widget_driver_generator.dart

```diff
@@ -30,30 +32,25 @@ class WidgetDriverGenerator extends GeneratorForAnnotation<GenerateTestDriver> {
     // Start - TestDriver generation
     //###################################
 
-    final testDriverClassName = '_\$Test$driverClassName';
+    final testDriverCodeProvider = TestDriverCodeProvider(
+      methods: visitor.methods,
+      properties: visitor.properties,
+      fields: visitor.fields,
+      driverClassName: driverClassName,
+    );
 
-    codeWriter.writeCode('class $testDriverClassName extends TestDriver implements $driverClassName {');
-    final annotationVisitor = AnnotationVisitor(codeWriter: codeWriter);
-    element.visitChildren(annotationVisitor);
-    codeWriter.writeCode('}');
+    codeWriter.writeCode(testDriverCodeProvider.code);
 
     //###################################
     // Start - DriverProvider generation
     //###################################
 
-    codeWriter.writeCode('class $providerClassName extends WidgetDriverProvider<$driverClassName> {');
+    final driverProviderCodeProvider = DriverProviderCodeProvider(
+      fields: visitor.providableFields,
+      driverClassName: driverClassName,
+    );
 
-    codeWriter.writeCode('@override');
-    codeWriter.writeCode('$driverClassName buildDriver(BuildContext context) {');
-    codeWriter.writeCode('return $driverClassName(context);');
-    codeWriter.writeCode('}');
-
-    codeWriter.writeCode('@override');
-    codeWriter.writeCode('$driverClassName buildTestDriver() {');
-    codeWriter.writeCode('return $testDriverClassName();');
-    codeWriter.writeCode('}');
-
-    codeWriter.writeCode('}');
+    codeWriter.writeCode(driverProviderCodeProvider.driverProviderClass);
```

Comment:
For the testDriverCodeProvider we say just provider.code… but here we say provider.driverProviderClass. Why the difference ?
Both provide code for the codeWriter right ?